### ISSUE TITLE
Feature Request: Inbox display of conversations

### ISSUE THREAD

**Issue by arena** [ProgressiveCoders/Spoke#1778]
**Problem**
It is difficult to see the replies you've gotten as a texter at a glance. It is also difficult to go back and find an interesting conversation. Often, as a texter, you'll want to handle a certain type of reply first (e.g. "yes, I'll vote. Where's my polling place??") but the only way to go through replies is one by one.

**Solution**
An inbox display organized by campaign that shows all the conversations, a preview of the most recent message, and their state (open, handled, bookmarked*, ..., etc)

**Context**
texter ui: dashboard

**Comment by ibrand** [comment]
@arena were you imagining something akin to the Warren/Thrutext sidebar? Or is this actually meant for the todo screen?

**Comment by arena** [comment]
@ibrand No, nothing like ThruText because I think that really slows down responding time and Spoke should stay efficient. I am imagining this for the ToDos screen

**Comment by arena** [comment]
I imagine something very much like Gmail or another email client with a way to sort by campaign ... needs some details and design but the user needs are:
- scan conversations for things that seem urgent
- scan conversations for things that need follow up
- archive conversations
- show what's new and unread (unhandled)
- see what's been bookmarked (starred)

Ideally, you would also be able search your conversations for a keyword or contact name. 

This would also open up the possibility for multi-select and then batching manual responses. e.g. let me take care of all these trolls first and then enjoy a session of responding.

**Comment by Frydafly** [comment]
Looping in @agreenspan24 who has done some really cool work in his fork of Spoke related to creating a texter inbox sidebox!

**Comment by agreenspan24** [comment]
Just also going to tag the Colorado version of the code, which is based off of Spoke v8: https://github.com/agreenspan24/Spoke/pull/27

We could use this as a starting point at least!

**Comment by arena** [comment]
It's interesting indeed, my main comments are that any improvements to navigating conversations will:
1. work on mobile
2. still support auto-advance for rapid responding (something that spoke does better than other apps)

**Comment by DinahSanders** [comment]
Querying whether there really needs to be bookmarking/starring. Wouldn't these message states be sufficient?

- unsent initial texts
- sent but no reply from contact [not shown in texter views of conversations]
- unread replies 
- unhandled replies (read but skipped)
- responded ('Past Messages' / archive)

If the default behavior after reading a message and responding is to move it to Past Messages (renamed 'Archive') 
and 
if the Archive is easily searchable (or at least viewable as a reverse chronological list showing the last reply, somewhat like Mobile Commons), 
doesn't that cover everything we need?
